python-api/app/integrations/openrouter/client.py

```
@@ -1,180 +1,440 @@
-        Sends a text completion request and parses response.
+        Convert an async generator into a sync generator by running the event loop internally.
+        WARNING: Use with caution; better to make the calling function async.
         """
-        response = self._post("/completions", payload.model_dump(exclude_none=True))
-        data = self._handle_response(response)
-        return CompletionResponse(**data)
+        # Use the existing event loop if available, otherwise create a new one
+        try:
+            loop = asyncio.get_running_loop()
+            # If there's already a running loop, we need to use a different approach
+            import concurrent.futures
```

Ruff (0.12.2)

`concurrent.futures` imported but unused

Remove unused import: `concurrent.futures`

(F401)